pkg/gofr/http/middleware/basic_auth_test.go

```diff
@@ -179,3 +180,144 @@ func Test_BasicAuthMiddleware_well_known(t *testing.T) {
+			authHeader:   "Basic " + base64.StdEncoding.EncodeToString([]byte("user")),
+			expectedUser: "",
+			expectedPass: "",
+			expectedOk:   false,
+		},
+		{
+			name:         "Empty Authorization Header",
+			authHeader:   "",
+			expectedUser: "",
+			expectedPass: "",
+			expectedOk:   false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			req := httptest.NewRequest(http.MethodGet, "/", http.NoBody)
+			req.Header.Set("Authorization", tc.authHeader)
+
+			username, password, ok := parseBasicAuth(req)
+
+			assert.Equal(t, tc.expectedOk, ok)
+			assert.Equal(t, tc.expectedUser, username)
+			assert.Equal(t, tc.expectedPass, password)
+		})
+	}
+}
+
+func TestRespondUnauthorized(t *testing.T) {
+	rr := httptest.NewRecorder()
+	respondUnauthorized(rr, "Test unauthorized message")
+
+	assert.Equal(t, http.StatusUnauthorized, rr.Code)
+	assert.Equal(t, "Unauthorized: Test unauthorized message\n", rr.Body.String())
+}
+
+func TestValidateCredentials(t *testing.T) {
+	validationFunc := func(user, pass string) bool {
+		return user == "validUser" && pass == "validPass"
+	}
+
+	validationFuncWithDB := func(_ *container.Container, user, pass string) bool {
+		return user == "dbUser" && pass == "dbPass"
+	}
+
+	provider := BasicAuthProvider{
+		Users:                       map[string]string{"storedUser": "storedPass"},
+		ValidateFunc:                validationFunc,
+		ValidateFuncWithDatasources: validationFuncWithDB,
+		Container:                   &container.Container{},
+	}
+
+	testCases := []struct {
+		name     string
+		username string
+		password string
+		expected bool
+	}{
+		{
+			name:     "Valid Credentials with ValidateFunc",
+			username: "validUser",
+			password: "validPass",
+			expected: true,
+		},
+		{
+			name:     "Valid Credentials with ValidateFuncWithDatasources",
+			username: "dbUser",
+			password: "dbPass",
+			expected: true,
+		},
+		{
+			name:     "Valid Credentials with Stored User",
+			username: "storedUser",
+			password: "storedPass",
+			expected: true,
+		},
+		{
+			name:     "Invalid Credentials",
+			username: "invalidUser",
+			password: "invalidPass",
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := validateCredentials(provider, tc.username, tc.password)
+			assert.Equal(t, tc.expected, result)
+		})
+	}
+}
+
+func TestBasicAuthMiddleware_NoAuthHeader(t *testing.T) {
+	authProvider := BasicAuthProvider{
+		Users: map[string]string{"user": "password"},
+	}
+
+	middleware := BasicAuthMiddleware(authProvider)
+	handler := http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	req := httptest.NewRequest(http.MethodGet, "/", http.NoBody)
+	rr := httptest.NewRecorder()
+	middleware(handler).ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusUnauthorized, rr.Code)
+	assert.Contains(t, rr.Body.String(), "Unauthorized: Authorization header missing")
```

Comment:
This test condition is failing as, on missing header the checks respond with - "Invalid or missing header"